feat(revenue-analytics): add virtual revenue properties for groups

rafaeelaudibert opened the pull request:
- Introduced virtual properties `$virt_revenue` and `$virt_revenue_last_30_days` for groups in the taxonomy.
- Create new group revenue analytics schema with join logic and tests to ensure proper functionality.
- Adjusted property definition API to include virtual properties for both persons and groups.
- Updated frontend settings to support group joins in revenue analytics.

Closes #35016
Closes #33505

rafaeelaudibert committed:
feat(revenue-analytics): add virtual revenue properties for groups

- Introduced virtual properties `$virt_revenue` and `$virt_revenue_last_30_days` for groups in the taxonomy.
- Create new group revenue analytics schema with join logic and tests to ensure proper functionality.
- Adjusted property definition API to include virtual properties for both persons and groups.
- Updated frontend settings to support group joins in revenue analytics.

Closes #35016
Closes #33505

rafaeelaudibert committed:
fix: Fix imports

rafaeelaudibert committed:
fix: Remove extra `persons` on tests name

It's obvious it's about persons

rafaeelaudibert committed:
feat: Include `groups_revenue_analytics` to the database

rafaeelaudibert committed:
tests: Enable strict mode by default when testing properties

This surfaces errors much more easily

rafaeelaudibert committed:
fix: Include `group_type_index` on group property

rafaeelaudibert committed:
fix: Use `group` on singular

That's the key, big mistake adding the plural

rafaeelaudibert committed:
tests: Remove unused snapshots

arthurdedeus reviewed (approved):
Looking good. Only weird part is the `_create_group` test util that seems not to be creating groups

rafaeelaudibert commented on posthog/taxonomy/property_definition_api.py:
```diff
@@ -541,6 +541,19 @@ class PropertyDefinitionViewSet(
+        {
+            "id": "$builtin_" + key,
+            "name": key,
+            "is_numerical": val["type"] == "Numeric",
```
I see `is_numerical` almost everywhere. You can see on line 429 on this file that `is_numerical` was introduced 4 years ago

rafaeelaudibert committed:
refactor: Remove unused method

rafaeelaudibert committed:
refactor: Remove web analytics-related virtual fields from groups

rafaeelaudibert committed:
fix: Properly connect with event groups

We need to connect with groups via the `$group_$` keys rather than going through the person - there's no connection between `person` and `group`s unfortunately.

rafaeelaudibert committed:
fix: fix mypy

`ast.Field` inherits from `ast.Expr` so I don't understand why this wouldn't be valid, but sure! let's fix it

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

github-actions[bot] committed:
Update query snapshots

rafaeelaudibert committed:
fix: Use proper field

Not sure why I changed it

github-actions[bot] committed:
Update query snapshots
